crates/cast/src/starknet_commands/account/delete.rs

```diff
@@ -0,0 +1,57 @@
+use anyhow::{anyhow, bail, Context, Result};
+use camino::Utf8PathBuf;
+use cast::helpers::response_structs::AccountDeleteResponse;
+use cast::parse_number;
+use clap::Args;
+use serde_json::Map;
+use serde_json::Value;
+
+#[derive(Args, Debug)]
+#[command(about = "Delete an account with all important secrets")]
+pub struct Delete {
+    /// Name of the account to be deployed
+    #[clap(short, long)]
+    pub name: Option<String>,
+
+    /// Custom open zeppelin contract class hash of declared contract
+    #[clap(long)]
+    pub network: Option<String>,
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn delete(name: &str, path: &Utf8PathBuf, network_name: &str) -> Result<AccountDeleteResponse> {
```

Comment:
let's maybe also add an option to remove the account from a Scarb.toml profile (if it exists and is indeed included). You can consult `account create` command to see how it is added using `add_profile` toggle, and do something similar (but reversed) here :)